torchvision/datasets/kitti.py

```diff
@@ -0,0 +1,171 @@
+import csv
+import os
+from typing import Any, Callable, List, Optional, Tuple
+from urllib.error import URLError
+
+from PIL import Image
+
+from .utils import download_and_extract_archive
+from .vision import VisionDataset
+
+
+class Kitti(VisionDataset):
+    """`KITTI <http://www.cvlibs.net/datasets/kitti>`_ Dataset.
+
+    Args:
+        root (string): Root directory where images are downloaded to.
+            Expects the following folder structure if download=False:
+
+            .. code::
+
+                <root>
+                    └── Kitti
+                        └─ raw
+                            ├── training
+                            |   ├── image_2
+                            |   └── label_2
+                            └── testing
+                                └── image_2
+        split (string): The dataset split to use. One of {``train``, ``test``}.
+            Defaults to ``train``.
+        transform (callable, optional): A function/transform that  takes in an PIL image
+            and returns a transformed version. E.g, ``transforms.ToTensor``
+        target_transform (callable, optional): A function/transform that takes in the
+            target and transforms it.
+        transforms (callable, optional): A function/transform that takes input sample
+            and its target as entry and returns a transformed version.
+        download (bool, optional): If true, downloads the dataset from the internet and
+            puts it in root directory. If dataset is already downloaded, it is not
+            downloaded again.
+
+    """
+
+    mirrors = [
+        "https://s3.eu-central-1.amazonaws.com/avg-kitti/",
+    ]
+    resources = [
+        "data_object_image_2.zip",
+        "data_object_label_2.zip",
+    ]
+    image_dir_name = "image_2"
+    labels_dir_name = "label_2"
+
+    def __init__(
+        self,
+        root: str,
+        split: str = None,
```

Comment:
Not sure if I missed anything, but why are we using `None` as default value here? Shouldn't it be `"train"`? Plus, if there are only two splits, `train: bool = True` might even be easier. If I'm wrong and we need `split = None` somewhere, we should at least fix the annotation:

```suggestion
        split: Optional[str] = None,
```